src/transformers/pipelines/base.py

```diff
@@ -1070,7 +1084,12 @@ def __call__(self, inputs, *args, num_workers=0, batch_size=1, **kwargs):
                 return outputs
             else:
                 return self.run_multi(inputs, preprocess_params, forward_params, postprocess_params)
-        elif Dataset is not None and isinstance(inputs, Dataset):
+        elif (
+            Dataset is not None
+            and isinstance(inputs, Dataset)
+            or isinstance(inputs, types.GeneratorType)
```

Comment:
```suggestion
            and isinstance(inputs, (Dataset, types.GeneratorType))
```
You can test for several types in the same check (or put parenthesis if the or was supposed to be interpreted differently)

Reply:
I know, wanted to keep them separate explicitly though because the `generator` support can be done also for `tf` (and `flax` which I am adding too)

Reply:
I changed to reflect that.